cpp/log/sqlite_db-inl.h

```diff
@@ -73,13 +91,50 @@ sqlite3* SQLiteOpen(const std::string& dbfile) {
 
 template <class Logged>
 SQLiteDB<Logged>::SQLiteDB(const std::string& dbfile)
-    : db_(SQLiteOpen(dbfile)), tree_size_(0) {
-  if (FLAGS_sqlite_synchronous_off) {
-    sqlite::Statement statement(db_, "PRAGMA synchronous = OFF");
+    : db_(SQLiteOpen(dbfile)),
+      tree_size_(0),
+      transaction_size_(0),
+      in_transaction_(false) {
+  std::unique_lock<std::mutex> lock(lock_);
+  {
+    std::ostringstream oss;
+    oss << "PRAGMA synchronous = " << FLAGS_sqlite_synchronous_mode;
+    sqlite::Statement statement(db_, oss.str().c_str());
     CHECK_EQ(SQLITE_DONE, statement.Step());
-    LOG(WARNING) << "SQLite \"synchronous\" pragma set to OFF";
+    LOG(WARNING) << "SQLite \"synchronous\" pragma set to "
+                 << FLAGS_sqlite_synchronous_mode;
+    if (FLAGS_sqlite_batch_into_transactions) {
+      LOG(WARNING) << "SQLite running with batched transactions, you should "
+                   << "set sqlite_synchronous_mode = FULL !";
+    }
   }
 
+  {
+    std::ostringstream oss;
+    oss << "PRAGMA journal_mode = " << FLAGS_sqlite_journal_mode;
+    sqlite::Statement statement(db_, oss.str().c_str());
```

Comment:
Ugh, ```sqlite::Statement``` should take a ```const string&``` for the statement, so you can build it easily, this is rather silly... For later, though.